cmd/limactl/main.go

```diff
@@ -215,6 +218,65 @@ func handleExitCoder(err error) {
 	}
 }
 
+// executeWithPluginSupport handles command execution with plugin support.
+func executeWithPluginSupport(rootCmd *cobra.Command, args []string) error {
+	if len(args) > 0 {
+		cmd, _, err := rootCmd.Find(args)
+		if err != nil || cmd == rootCmd {
+			return runExternalPlugin(rootCmd.Context(), args[0], args[1:])
```

Comment:
I'm sorry, I just realized that we need to differentiate between "plugin not found" and "plugin has been executed, but returned an error".

With the current implementation we get:

```console
❯ rm ~/bin/limactl-hello

❯ limactl hello
FATA[0000] exec: "limactl-hello": executable file not found in $PATH
```

But we really want:

```console
❯ limactl hello
FATA[0000] unknown command "hello" for "limactl"

Did you mean this?
	shell
```

So we should change the code to:

```suggestion
			// Function calls os.Exit() if it found and executed the plugin
			runExternalPlugin(rootCmd.Context(), args[0], args[1:])
```

Reply:
now works a expected.

```
➜  lima git:(allow-cmd-alias-via-path-lookup) ✗ limactl sh 
Running limactl-shell alias with arguments: 
FATA[0000] "limactl shell" requires at least 1 arg(s), only received 0.
See 'limactl shell --help'.

Usage:  limactl shell [flags] INSTANCE [COMMAND...]

Execute shell in Lima 
```